src/external-actor-on-team-approved-pr.ts

```diff
@@ -0,0 +1,101 @@
+import * as core from "@actions/core";
+import * as httpm from "@actions/http-client";
+import { context, getOctokit } from "@actions/github";
+
+process.on("unhandledRejection", handleError);
+main().catch(handleError);
+
+async function main(): Promise<any> {
+  const http: httpm.HttpClient = new httpm.HttpClient(
+    "external-actor-on-team-approved-pr"
+  );
+
+  if (context.payload.pull_request == null) {
+    console.log("No pull request found in the context");
+    return;
+  }
+
+  const apiToken = core.getInput("api-token", { required: true });
+  const slackWebhookUrl = core.getInput("slack-webhook-url", {
+    required: true,
+  });
+
+  const github = getOctokit(apiToken, {});
+
+  // Get the team members
+  const teamSlug = core.getInput("team-slug", { required: true });
+  const { data: members } = await github.rest.teams.listMembersInOrg({
+    org: context.repo.owner,
+    team_slug: teamSlug,
+  });
+  const teamMemberLogins = members.map((member) => member.login);
+  console.log(`Team members retrieved: ${teamMemberLogins.join(", ")}`);
+
+  // If the actor is a team member,
+  // skip the Slack notification
+  const actor = context.actor;
+  if (teamMemberLogins.includes(actor)) {
+    console.log(
+      `Actor is a team member (${actor}), skipping Slack notification.`
+    );
+    return;
+  }
+
+  // Get the list of reviewers who have approved the PR
+  const { data: reviews } = await github.rest.pulls.listReviews({
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    pull_number: context.payload.pull_request.number,
+  });
+
+  let isApproved = false;
+
+  // Check whether any of the team members
+  // have approved the PR
+  for (const review of reviews) {
+    if (review.state === "APPROVED") {
+      if (review.user == null) {
+        continue;
+      }
+      const approverLogin = review.user.login;
+      if (teamMemberLogins.includes(approverLogin)) {
+        isApproved = true;
+        console.log(
+          `The PR is approved by a team member: ${approverLogin}. Notifying Slack ...`
+        );
+        break;
+      }
+    }
+  }
+
+  // If no one on the team has approved it yet,
+  // skip the Slack notification
+  if (!isApproved) {
+    console.log(
+      "No team members have approved the PR yet, skipping Slack notification."
+    );
```

Comment:
Nit: (feel free to ignore) Would be nice to use the logging provided in the core package: https://github.com/actions/toolkit/tree/main/packages/core#logging